File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
        int res = 0 , ans = 0;
        vector<int> helper(k) ;
        helper[0] = 1 ;
        for(auto  &i: nums){
            res =  (res+i%k +k)%k;
            ans +=  helper[res];
            helper[res]++;
        }
        return ans;
    }
};

File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        res  =  Counter()
        ans =  0 
        hel = 0
        for i in nums :
            hel += i
            if hel%k == 0 :
                ans += 1
            ans += res[hel%k]
            res[hel%k] += 1 
                 
        
        return ans